src/shared/user-menu.jsx

```diff
@@ -173,7 +112,19 @@ function mapStateToProps(state) {
   return {
     isAuthenticated,
     avatar: state.userData.avatar,
+    name: state.userData.name,
+  }
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    login: () => {
+      dispatch(login())
+    },
+    logout: () => {
+      dispatch(logout())
+    },
```

Comment:
@wlach  we can make the common code for login not for logout because in response of logout we are doing different things in src/actions/action.js (we are doing dispatch things) where as in user-menu.jsx( we are doing setState things ). Any suggestions regarding this.

Reply:
I think by providing a callback function you can do what you need here. E.g.:

```js
function logout(successCallback, errorCallback) {
  fetch('/logout/').then(response => {
    if (response.ok) {
      successCallback()
      return
    } 
    return errorCallback()
  });
}
```

There might be an even better way of doing this, but I think this will work.

Reply:
thanks :)